docs/rpc-endpoints.md

```diff
@@ -0,0 +1,255 @@
+      "access": "public",
+      "args": [],
+      "outputs": {
+        "type": {
+          "response": {
+            "ok": "bool",
+            "error": "none"
+          }
+        }
+      }
+    },
+    {
+      "name": "get-exotic-data-info",
+      "access": "read_only",
+      "args": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "outputs": {
+        "type": {
+          "tuple": [
+            {
+              "name": "btc-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "burn-block-time",
+              "type": "uint128"
+            },
+            {
+              "name": "id-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-miner",
+              "type": "principal"
+            },
+            {
+              "name": "vrf-seed",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "variables": [],
+  "maps": [
+    {
+      "name": "block-data",
+      "key": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "value": [
+        {
+          "name": "btc-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "burn-block-time",
+          "type": "uint128"
+        },
+        {
+          "name": "id-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-miner",
+          "type": "principal"
+        },
+        {
+          "name": "vrf-seed",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "fungible_tokens": [],
+  "non_fungible_tokens": []
+}
+```
+
+### GET /v2/contracts/source/[Stacks Address]/[Contract Name]
+
+Fetch the source for a smart contract. Returned as a JSON string.
+
+### POST /v2/contracts/call-read/[Stacks Address]/[Contract Name]/[Function Name]
+
+Call a read-only public function on a given smart contract.
+
+The smart contract and function are specified using the URL path. The arguments and
+the simulated `tx-sender` are supplied via the POST body in the following JSON format:
```

Comment:
Am I the only one feeling ambivalent about this endpoint taking a sender as an argument?

Reply:
Since this endpoint is doing a Clarity function call, we need to fill in the global environment variables.  Related, shouldn't this payload also take `block-height`?

Reply:
Right `tx-sender` needs to be filled in -- we could pick a default for it, and allow overriding it via a query string or something.

As for block-height -- no, that doesn't need to be set. That's just the height of the chain tip that it's being evaluated at (the call is evaluated as if it were the first transaction in a block following the chain tip). When we add a block-hash specified endpoint, then the read-only call would be evaluated as if it were the first transaction in a new block following that block-hash.

Reply:
`tx-sender` is prefixed by `tx`. 
From a smart contract developer point of view, I'm going to use `tx-sender` with the assumption that this will return me the principal of the sender who signed this transaction.
In the context of this RPC-call, the code `tx-sender` will be evaluated without providing this guarantee.

Should `tx-sender` becomes an optional `tx-sender?` ? Should this endpoint require a signature deduct the sender from this? Should we rename `tx-sender`? Or just make this clear in the `tx-sender` documentation?

Reply:
> Should tx-sender becomes an optional tx-sender? ? Should this endpoint require a signature deduct the sender from this? Should we rename tx-sender? Or just make this clear in the tx-sender documentation?

I think it's more cumbersome for tx-sender to be an optional type in Clarity, but I agree with you that the variable is a little strange in the context of the API read-only call. I think this is something we may want to make more clear in the API doc, and should consider making default to something like a null principal. I'll open an issue for it.